app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -221,40 +251,41 @@ class ArticleEditDetailsFragment : Fragment(), WatchlistExpiryDialog.Callback, L
     }
 
     private fun setLoadingState() {
-        binding.progressBar.visibility = VISIBLE
-        binding.usernameButton.visibility = INVISIBLE
-        binding.thankButton.visibility = INVISIBLE
-        binding.editComment.visibility = INVISIBLE
-        binding.diffText.visibility = INVISIBLE
-        binding.diffCharacterCountView.visibility = INVISIBLE
-    }
-
-    private fun updateUI() {
-        binding.diffText.scrollTo(0, 0)
-        binding.diffText.text = ""
-        binding.usernameButton.text = currentRevision!!.user
-        binding.editTimestamp.text = DateUtil.getDateAndTimeWithPipe(DateUtil.iso8601DateParse(currentRevision!!.timeStamp))
-        binding.editComment.text = currentRevision!!.comment
-        binding.newerIdButton.isClickable = newerRevisionId != -1L
-        binding.olderIdButton.isClickable = olderRevisionId != 0L
-        setEnableDisableTint(binding.newerIdButton, newerRevisionId == -1L)
-        setEnableDisableTint(binding.olderIdButton, olderRevisionId == 0L)
+        binding.progressBar.isVisible = true
+        binding.revisionDetailsView.isVisible = false
+        binding.diffRecyclerView.isVisible = false
+    }
+
+    private fun updateAfterRevisionFetchSuccess() {
+        if (viewModel.revisionFrom != null) {
```

Comment:
Is it possible to use `viewModel.revisionFrom?.let` so that we don't need to add `!!` in the following places where we use it.

Reply:
yep!